src/renderer/components/detailed-view-components/DetailedViewCore.tsx

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+
+import React, { useMemo, useState } from 'react';
+import { HTMLTable, Switch } from '@blueprintjs/core';
+import { ComputeNode } from '../../../data/GraphOnChip';
+import LinkDetails from '../LinkDetails';
+import MemoryPlotRenderer from '../memory-renderer/MemoryPlotRenderer';
+import { L1RenderConfiguration, getChartData } from '../memory-renderer/PlotConfigurations';
+import { MemoryLegendElement } from '../memory-renderer/MemoryLegendElement';
+import { formatSize } from '../../utils/numbers';
+
+interface DetailedViewCoreRendererProps {
+    node: ComputeNode;
+    temporalEpoch: number;
+}
+
+const DetailedViewCoreRenderer: React.FC<DetailedViewCoreRendererProps> = ({ node, temporalEpoch }) => {
+    const MINIMAL_MEMORY_RANGE_OFFSET = 0.998;
+    const [zoomedInView, setZoomedInView] = useState(false);
+
+    // TODO: we need to find a creative way to render CONSUMED size vs SIZE - this might become interesting metrics to show
+    // TODO: maybe a toggle to flip between the two with a ghosting allocated size overlay
+
+    // usage should always be false
+    const dataBuffers = getChartData(node.coreL1Memory.dataBuffers, false);
+    const dataBuffersConfig = { ...L1RenderConfiguration };
+    dataBuffersConfig.title = 'Data Buffers';
+    const dataBufferZoomRangeStart = useMemo(
+        () => (node.coreL1Memory.dataBuffers[0]?.address ?? 0) * MINIMAL_MEMORY_RANGE_OFFSET,
+        [node.coreL1Memory.dataBuffers],
+    );
+
+    const dataBufferZoomRangeEnd = useMemo(() => {
+        const lastItem = node.coreL1Memory.dataBuffers.at(-1);
+        const lastAddress = lastItem?.address ?? 0;
+        const lastSize = lastItem?.size ?? 0;
+
+        return (lastAddress + lastSize) * (1 / MINIMAL_MEMORY_RANGE_OFFSET) || node.coreL1Memory.l1Size;
+    }, [node.coreL1Memory.dataBuffers, node.coreL1Memory.l1Size]);
+
+    const binaryBuffers = getChartData(node.coreL1Memory.binaryBuffers, true);
+    const binaryBuffersConfig = { ...L1RenderConfiguration };
+    binaryBuffersConfig.title = 'Binary Buffers';
+    const binaryBufferZoomRangeStart = useMemo(
+        () => (node.coreL1Memory.binaryBuffers[0]?.address ?? 0) * MINIMAL_MEMORY_RANGE_OFFSET,
+        [node.coreL1Memory.binaryBuffers],
+    );
+
+    const binaryBufferZoomRangeEnd = useMemo(() => {
```

Comment:
does this really need a useMemo?

Reply:
I think it won't be a problem as it is not a very expensive computation.
But without it, the computation would happen every time the user opens the details view and they click the zoom switch.

Reply:
it a loop over 20 item list max